### ISSUE TITLE
<format>: `format_to(back_inserter(string), ...)` should reserve space.

### ISSUE THREAD

**Issue by sylveon** [microsoft/STL#1916]
Much like the special treatment of string added in #1894, `format_to` should try to detect that a string is passed and reserve `string.size() + _Fmt.size() + _Args._Estimate_required_capacity()`. This would make `format_to` a string just as fast as `format`

**Comment by barcharcraz** [comment]
Is this still a valid issue? It appears to me that foirmat_to will activate the same optimization now thanks to your fast path in the iterator buffer.

**Comment by sylveon** [comment]
Yes, this is still valid. The estimated capacity is only reserved when calling vformat (and by extension format). format_to and vformat_to do not reserve space:
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/format#L2894-L2899

**Comment by barcharcraz** [comment]
Ahh I see, Right now it will reserve enough space for the whole 256 character small buffer each time the iterator buffer needs to flush, but it won't reserve the whole thing in advance.

Benchmarking will need to get done as "estimate required capacity" involves actually running through all the formatting code and discarding the output.

**Comment by CaseyCarter** [comment]
> Benchmarking will need to get done as "estimate required capacity" involves actually running through all the formatting code and discarding the output.

It's not that bad:
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/format#L1496-L1512
